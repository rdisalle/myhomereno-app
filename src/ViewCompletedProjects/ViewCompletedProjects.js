import React, { Component } from 'react';
import './ViewCompletedProjects.css';
import ProjectPage from '../ProjectPage/ProjectPage';
import { Link, NavLink } from 'react-router-dom';
import MyHomeRenoContext from '../MyHomeRenoContext';
import PropTypes from 'prop-types';

class ViewCompletedProjects extends Component {

    static contextType = MyHomeRenoContext;

    countEstimatesForProject = (estimates=[], projectId) =>
    estimates.filter(estimate => estimate.project_id === projectId).length;

    onDelete = () => {
        this.props.history.push('/view-current-project-list');
      };

    render() {
        const { projects=[], estimates=[] } = this.context;

        return (
        <div className="ViewCurrentProjects">
            <div className="ViewProjects_select">
                <Link className="ViewProjectType_link" to={'/view-current-project-list'} >
                    All Projects
                </Link>
                <Link className="ViewProjectType_link" to={'/view-not-started-project-list'}>
                    Not Started
                </Link>
                {' '}
                <Link className="ViewProjectType_link" to={'/view-in-progress-project-list'}>
                    In Progress
                </Link>
                {' '}
                <NavLink className="ViewProjectType_link" activeClassName="active" to={'/view-completed-project-list'}>
                    Completed
                </NavLink>
            </div>
                <Link className="AddProject_link" to={'/add-project'}>
                    Add Project
                </Link>
            <h1 className= "ViewCurrentProjects_header">Completed Projects</h1>
            <ul className="ViewCurrentProjects_list">
                    {projects.filter(project => project.status === 'Completed').map(project => (

                    <li className="ViewCurrentProjects_project" key={project.id}> 
                    <Link
                        className="ViewCurrentProjects_project-link"
                        to={`/view-project-page/${project.id}`}
                    >
                        {project.name}
                    </Link>
                    <ProjectPage className="ProjectList_data"
                        id={project.id}
                        summary={project.summary}
                        estimated_cost={project.estimated_cost}
                        room={project.room}
                        details={project.details}
                        total_time={project.total_time}
                        type={project.type}
                        status={project.status}
                        date_created={project.date_created}
                        delete= {this.onDelete}
                    />
                    <div className="ViewProjectList_estimate">
                    <span className="ViewProjectList_num-scores">
                        Estimates: ({this.countEstimatesForProject(estimates, project.id)});
                    </span>
                    </div>
                    </li>
                ))}
            </ul>
        </div>
        );
    };
};

ViewCompletedProjects.propTypes = {
    history: PropTypes.object
  };
  
export default ViewCompletedProjects;
                    
                
