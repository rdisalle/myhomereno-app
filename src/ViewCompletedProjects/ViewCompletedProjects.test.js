import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ViewCompletedProjects from './ViewCompletedProjects';

describe(`ViewCompletedProjects component`, () => {
  const props = {
    projects: [
      {
        "id": "cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1",
        "name": 'My Project 1',
        "estimated_cost": '$10,000',
        "total_time": '4 weeks',
        "room":'Bathroom',
        "details": 'Entire bath reno',
        "type": 'Contactor',
        "status": 'Completed',
        "summary": 'Getting bathroom redone',
      },
      {
        "id": "cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1",
        "name": 'My Project 2',
        "estimated_cost": '$10,000',
        "total_time": '4 weeks',
        "room":'Bathroom',
        "details": 'Entire bath reno',
        "type": 'Contactor',
        "status": 'Completed',
        "summary": 'Getting bathroom redone',
      }
    ]
  };

  it('renders a .ViewCompletedProjects by default', () => {
    const wrapper = shallow(<ViewCompletedProjects />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders a Project in ul for each project in array', () => {
    const ul = shallow(<ViewCompletedProjects {...props} />)
      .find('ul');
    expect(toJson(ul)).toMatchSnapshot();
  });
});