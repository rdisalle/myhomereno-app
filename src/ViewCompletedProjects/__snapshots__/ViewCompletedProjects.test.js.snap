// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewCompletedProjects component renders a .ViewCompletedProjects by default 1`] = `
<div
  className="ViewCurrentProjects"
>
  <div
    className="ViewProjects_select"
  >
    <Link
      className="ViewProjectType_link"
      to="/view-current-project-list"
    >
      All Projects
    </Link>
    <Link
      className="ViewProjectType_link"
      to="/view-not-started-project-list"
    >
      Not Started
    </Link>
     
    <Link
      className="ViewProjectType_link"
      to="/view-in-progress-project-list"
    >
      In Progress
    </Link>
     
    <NavLink
      activeClassName="active"
      className="ViewProjectType_link"
      to="/view-completed-project-list"
    >
      Completed
    </NavLink>
  </div>
  <Link
    className="AddProject_link"
    to="/add-project"
  >
    Add Project
  </Link>
  <h1
    className="ViewCurrentProjects_header"
  >
    Completed Projects
  </h1>
  <ul
    className="ViewCurrentProjects_list"
  />
</div>
`;

exports[`ViewCompletedProjects component renders a Project in ul for each project in array 1`] = `
<ul
  className="ViewCurrentProjects_list"
/>
`;
